implentio/ui-toolkit
# Feat IUT-39/menu group

## Description
This is the component used in the navigation bar which groups menu items together. It can include single navigation items and grouped navigation items. Grouped navigation items have a collapsed subitem group which expands on click. Clicking subNav items will navigate a page. For single nav items, clicking on it will navigate to a page
NOTE: active state does not change right now because it requires a parent to pass down the state

## Link to JIRA ticket
https://implentio.atlassian.net/jira/software/projects/IUT/boards/10?selectedIssue=IUT-39

## Applicable Screenshots
![image](https://github.com/implentio/ui-toolkit/assets/93560211/9e394fe6-be40-42dd-8321-f34211303c83)

## Testing Instructions
- Fetch and checkout my branch `git fetch && git checkout <feat/IUT-39/MenuGroup>`
- npm run storybook into the CLI
- Check the following behaviors for the MenuGroup:

1. Clicking on grouped navitems will expand a subnav group and does not trigger an action
3. clicking a subnav item will trigger an action (it will not update active state!)
4. clicking a single (does not have a chevron) navItem will trigger an action  (it will not update active state!)
5. hovering will change text and chevron to purple (hover color) from blackish (default color)
6. in the GroupActive story, the action group has purple text and its subnav item has purple text and a purple bg
7. in the singleActive story, the navitem has purple text and bg
8. when toggling the MenuGroup type ("single" | "multiple"), single will only allow one grouped item to be expanded
9. if type is multiple, you can have multiple nav groups open
10. change active state through storybook controls and see if styling matches

## Checklist:
Check what is applicable by putting an 'x' in between the []:
